src/Maps/MapData.h

```diff
@@ -14,16 +14,45 @@
 
 enum class CellType;
 
+namespace Stream {
+	class Writer;
+	class Reader;
+	class SeekableReader;
+}
+
 // FILE FORMAT DOCUMENTATION:
 //     Outpost2SVN\OllyDbg\InternalData\FileFormat SavedGame and Map.txt.
 
 // ALT IMPLEMENTATION (with COM support)
 //     Outpost2SVN\MapEditor\OP2Editor.
 
-//Represents the data stored in an Outpost 2 map file.
-struct MapData
+//An Outpost 2 map file.
+class MapData
 {
-	MapHeader header;
+public:
+	MapData();
+	MapData(const MapHeader& mapHeader);
```

Comment:
I'm not sure this constructor should be public. It doesn't fully initialize the object on it's own. In particular, there is no tiles array, nor any of the tileset info. This would effectively construct an incomplete object that was not fully usable, and may violate certain class invariants (such as the size of the tileset array corresponding with the dimensions). This constructor might be best kept as private.

Reply:
I hadn't thought about making a private constructor. Agreed that the object wouldn't be fully usable. By removing this constructor, an outside entity is now unable to construct a MapData. We can decide what this process looks like in a different branch. 

Check out the solution I came up with. It isn't a private constructor, but does remove public use of MapHeader from MapData. I didn't want to make MapHeader definition a part of MapData just to keep from polluting the MapData header file with MapHeader definition if it could be helped. Guess if it is the right thing to do we can though.

-Brett